flask_resty/pagination.py

```diff
@@ -537,16 +582,39 @@ class RelayCursorPagination(CursorPaginationBase):
     """
 
     def get_page(self, query, view):
-        query, field_orderings = self.ensure_query_sorting(query, view)
+        (
+            sorting_field_orderings,
+            missing_field_orderings,
+        ) = self.get_sorting_and_missing_field_orderings(view)
+
+        field_orderings = sorting_field_orderings + missing_field_orderings
+
+        cursor_in, reversed = self.get_request_cursor(view, field_orderings)
+
+        if reversed:
+            field_orderings = self.reverse_field_ordering(field_orderings)
+
+            query = query.order_by(None)  # clear any existing order_by
+            query = view.sorting.sort_query_by_fields(
+                query, view, field_orderings
+            )
```

Comment:
aight this feels bad. Sorting via the `Sorting` class is done prior to `get_page` so there is already an `order_by` on the query, then we add some extra order-by's in based on `id` if they aren't already included. It feels unnecessary...the Pagination really needs to own the entire order_by. 

For the reverse case we actually need to remove the existing order by, reverse it and then add it back. I suspect that we should fully replace the query sort in all cases but not confident it won't break something. DRF seems to take this approach as well

Reply:
i am not sure why we need to change the order by? the order, whether you use before or after, shouldn't change...

Reply:
it's reversed b/c i don't know a way of doing a reverse LIMIT, so to get the previous page, we reverse the order by and flip the WHERE so instead of:
```sql
SELECT * 
FROM foo
WHERE id > 4 
ORDER BY id
LIMIT 10
```
You get 
```sql
SELECT * 
FROM foo
WHERE id < 4 
ORDER BY DESC id
LIMIT 10
```

Reply:
of course. that is annoying.

Reply:
isn't probably better if it's the sorting that tries to be aware of the pagination, and in that case flip it? or alternatively sorting calling into some pagination method to manipulate the sorting before it's applied?

Reply:
it's good question...maybe? Do you have an API in mind, I have a hard time knowing what is "best" in this library :P 

Isn't weird for encapsulation that these view properties call into each other all the time?

Reply:
a little. however pagination and sorting are coupled by nature, it's not extrinsic complexity.

I think something like

```py
def sort_query(self, query, view):
        field_orderings = self.get_request_field_orderings(view)
        if view.pagination:
            field_orderings = view.pagination.adjust_sorting(field_orderings)
        return self.sort_query_by_fields(query, view, field_orderings)
```

or even:

```py
def sort_query(self, query, view):
        field_orderings = self.get_request_field_orderings(view)
        if view.pagination and view.pagination.reverse_sorting:
            field_orderings = field_orderings[::-1]
        return self.sort_query_by_fields(query, view, field_orderings)
```